### PR TITLE
Convert to TF 0.12. Add tests. Add Codefresh test pipeline

### PR DESCRIPTION
**Pull request by goruha**
## what
* Port module to Terraform 0.12
* Pin all providers
* Add example for testing
* Add `bats` and `terratest` for the example
* Add Codefresh badge to point to the test pipeline in `terraform-modules` project
* Update README 

## why
* Module currently does not work with 0.12. Much easier syntax
* Better regression control
* Automatically test the example on every commit and pull request
* Provision resources on AWS in the test account and check the outputs for the correct values
* `terraform-modules` project contains pipelines for all terraform modules

## related
* Closes #27
* Closes #24 
* Closes #16

### CODE REVIEW

**Comment by goruha** [comment]
codefresh run test

**Comment by goruha** [comment]
codefresh run test

**Comment by goruha** [comment]
codefresh run test

**Comment by goruha** [comment]
codefresh run test

**Comment by goruha** [comment]
codefresh run test

**Comment by goruha** [comment]
codefresh run test

**Comment by goruha** [comment]
@aknysh addressed

File: examples/complete/fixtures.us-west-1.tfvars [hunk 1]
```diff
@@ -0,0 +1,7 @@
+region = "us-west-1"
```
Comment (aknysh) [inline examples/complete/fixtures.us-west-1.tfvars]:
```suggestion
region = "us-east-2"
```

and change the file name to `examples/complete/fixtures.us-east-2.tfvars`

File: test/src/examples_complete_test.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// Test the Terraform module in examples/complete using Terratest.
+func TestExamplesComplete(t *testing.T) {
+	t.Parallel()
+
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: "../../examples/complete",
+		Upgrade:      true,
+		// Variables to pass to our Terraform code using -var-file options
+		VarFiles: []string{"fixtures.us-west-1.tfvars"},
```
Comment (aknysh) [inline test/src/examples_complete_test.go]:
```suggestion
		VarFiles: []string{"fixtures.us-east-2.tfvars"},
```

**Pull request merged**